homeassistant/components/recorder/purge.py

```diff
@@ -112,6 +115,223 @@ def _select_event_state_and_attributes_ids_to_purge(
     return event_ids, state_ids, attributes_ids
 
 
+def _state_attrs_exist(attr: int | None) -> Select:
+    """Check if a state attributes id exists in the states table."""
+    return select(func.min(States.attributes_id)).where(States.attributes_id == attr)
+
+
+def _generate_find_attr_lambda(
+    attr1: int,
+    attr2: int | None,
+    attr3: int | None,
+    attr4: int | None,
+    attr5: int | None,
+    attr6: int | None,
+    attr7: int | None,
+    attr8: int | None,
+    attr9: int | None,
+    attr10: int | None,
+    attr11: int | None,
+    attr12: int | None,
+    attr13: int | None,
+    attr14: int | None,
+    attr15: int | None,
+    attr16: int | None,
+    attr17: int | None,
+    attr18: int | None,
+    attr19: int | None,
+    attr20: int | None,
+    attr21: int | None,
+    attr22: int | None,
+    attr23: int | None,
+    attr24: int | None,
+    attr25: int | None,
+    attr26: int | None,
+    attr27: int | None,
+    attr28: int | None,
+    attr29: int | None,
+    attr30: int | None,
+    attr31: int | None,
+    attr32: int | None,
+    attr33: int | None,
+    attr34: int | None,
+    attr35: int | None,
+    attr36: int | None,
+    attr37: int | None,
+    attr38: int | None,
+    attr39: int | None,
+    attr40: int | None,
+    attr41: int | None,
+    attr42: int | None,
+    attr43: int | None,
+    attr44: int | None,
+    attr45: int | None,
+    attr46: int | None,
+    attr47: int | None,
+    attr48: int | None,
+    attr49: int | None,
+    attr50: int | None,
+    attr51: int | None,
+    attr52: int | None,
+    attr53: int | None,
+    attr54: int | None,
+    attr55: int | None,
+    attr56: int | None,
+    attr57: int | None,
+    attr58: int | None,
+    attr59: int | None,
+    attr60: int | None,
+    attr61: int | None,
+    attr62: int | None,
+    attr63: int | None,
+    attr64: int | None,
+    attr65: int | None,
+    attr66: int | None,
+    attr67: int | None,
+    attr68: int | None,
+    attr69: int | None,
+    attr70: int | None,
+    attr71: int | None,
+    attr72: int | None,
+    attr73: int | None,
+    attr74: int | None,
+    attr75: int | None,
+    attr76: int | None,
+    attr77: int | None,
+    attr78: int | None,
+    attr79: int | None,
+    attr80: int | None,
+    attr81: int | None,
+    attr82: int | None,
+    attr83: int | None,
+    attr84: int | None,
+    attr85: int | None,
+    attr86: int | None,
+    attr87: int | None,
+    attr88: int | None,
+    attr89: int | None,
+    attr90: int | None,
+    attr91: int | None,
+    attr92: int | None,
+    attr93: int | None,
+    attr94: int | None,
+    attr95: int | None,
+    attr96: int | None,
+    attr97: int | None,
+    attr98: int | None,
+    attr99: int | None,
+    attr100: int | None,
+) -> StatementLambdaElement:
+    """Generate the find attributes select only once.
+
+    https://docs.sqlalchemy.org/en/14/core/connections.html#quick-guidelines-for-lambdas
+    """
+    return lambda_stmt(
```

Comment:
The `lambda_stmt` is actually looking at the code inside the lambda to generate the cache key so it seems like we are stuck writing this out if we want a single cache key with bindparams